examples/esplora_backend_with_tor.rs

```diff
@@ -0,0 +1,173 @@
+
+    let socks_addr = start_tor();
+
+    let client = Builder::new(esplora_url)
+        .proxy(&format!("socks5://{}", socks_addr))
+        .timeout(120) // seconds
+        .build_blocking()
+        .expect("Should never fail with no proxy and timeout");
+
+    println!(
+        "Connecting to {} via SOCKS5 proxy {}",
+        &esplora_url, &socks_addr
+    );
+
+    let blockchain = EsploraBlockchain::from_client(client, 20);
+
+    let wallet = create_wallet(network, &xpriv);
+
+    wallet.sync(&blockchain, SyncOptions::default()).unwrap();
+
+    let address = wallet.get_address(AddressIndex::New).unwrap().address;
+
+    println!("address: {}", address);
+
+    let balance = wallet.get_balance().unwrap();
+
+    println!("Available coins in BDK wallet : {} sats", balance);
+
+    if balance.confirmed > 10500 {
+        // the wallet sends the amount to itself.
+        let recipient_address = wallet
+            .get_address(AddressIndex::New)
+            .unwrap()
+            .address
+            .to_string();
+
+        let amount = 9359;
+
+        let tx = build_signed_tx(&wallet, &recipient_address, amount);
+
+        blockchain.broadcast(&tx).unwrap();
+
+        println!("tx id: {}", tx.txid());
+    } else {
+        println!("Insufficient Funds. Fund the wallet with the address above");
+    }
+
+    let export = FullyNodedExport::export_wallet(&wallet, "exported wallet", true)
+        .map_err(ToString::to_string)
+        .map_err(bdk::Error::Generic)
+        .unwrap();
+
+    println!("------\nWallet Backup: {}", export.to_string());
+}
+
+pub fn start_tor() -> String {
+    let socks_port = 19050;
+
+    let data_dir = format!("{}/{}", env::temp_dir().display(), "bdk-tor-example");
+    let log_file_name = format!("{}/{}", &data_dir, "log");
+
+    println!("Staring Tor in {}", &data_dir);
+
+    truncate_log(&log_file_name);
+
+    Tor::new()
+        .flag(TorFlag::DataDirectory(data_dir.into()))
+        .flag(TorFlag::LogTo(
+            LogLevel::Notice,
+            LogDestination::File(log_file_name.as_str().into()),
+        ))
+        .flag(TorFlag::SocksPort(socks_port))
+        .flag(TorFlag::Custom("ExitPolicy reject *:*".into()))
+        .flag(TorFlag::Custom("BridgeRelay 0".into()))
+        .start_background();
+
+    let mut started = false;
+    let mut tries = 0;
+    while !started {
+        tries += 1;
+        if tries > 120 {
+            panic!(
+                "It took too long to start Tor. See {} for details",
+                &log_file_name
+            );
+        }
+
+        thread::sleep(Duration::from_millis(1000));
+        started = find_string_in_log(&log_file_name, &"Bootstrapped 100%".into());
+    }
+
+    println!("Tor started");
+
+    format!("127.0.0.1:{}", socks_port)
+}
+
+fn truncate_log(filename: &String) {
+    let path = std::path::Path::new(filename);
+    if path.exists() {
+        let file = File::options()
```

Comment:
This will not work in `rust 1.56.1`.. You need to use `std::fs::OpenOption`.. Thats the error in CI..